src/main/java/emissary/place/Main.java

```diff
@@ -1040,33 +1042,71 @@ public void printPayload(IBaseDataObject payload) {
     }
 
     /**
-     * Split the output to separate files. This is for doing followon processing on the data, so just the data gets output
+     * Split the output to separate files. This is for doing follow on processing on the data, so just the data gets output
      * here. One payload or attachment per file. Directories will be created as needed underneath the baseOutputDir (can be
-     * specified with -d)
+     * specified with -d).
+     * <p>
+     * Note: Output will be suppressed if an IBDO form or filename would yield an output path outside the configured
+     * {@link #getBaseOutputDir() base output directory} path
+     * </p>
      *
      * @param payload the processed payload
      * @param att the list of attachments
      */
     public void handleSplitOutput(IBaseDataObject payload, List<IBaseDataObject> att) {
         String fn = getBaseOutputDir() + "/" + payload.shortName() + "." + payload.currentForm();
-        boolean status = Executrix.writeDataToFile(payload.data(), fn);
-        if (status) {
-            logger.debug("Wrote output to " + fn);
-        } else {
-            logger.error("Could not write output to " + fn);
-        }
 
+        String safePath;
+        try {
+            safePath = filePathIsWithinBaseDirectory(getBaseOutputDir(), fn);
+            if (Executrix.writeDataToFile(payload.data(), safePath)) {
+                logger.debug("Wrote output to {}", safePath);
+            } else {
+                logger.error("Could not write output to {}", safePath);
+            }
+        } catch (IllegalArgumentException e) {
+            logger.error("Could not write output to {}", fn, e);
+        }
         for (IBaseDataObject part : att) {
             fn = getBaseOutputDir() + "/" + part.shortName() + "." + part.currentForm();
-            status = Executrix.writeDataToFile(part.data(), fn);
-            if (status) {
-                logger.debug("Wrote attachment output to " + fn);
-            } else {
-                logger.error("Could not write output to " + fn);
+            try {
+                safePath = filePathIsWithinBaseDirectory(getBaseOutputDir(), fn);
+                if (Executrix.writeDataToFile(payload.data(), safePath)) {
+                    logger.debug("Wrote attachment output to " + safePath);
+                } else {
+                    logger.error("Could not write output to " + safePath);
+                }
+            } catch (IllegalArgumentException e) {
+                logger.error("Could not write output to {}", fn, e);
             }
         }
     }
 
+    /**
+     * Normalizes the provided paths and tests to see whether the file path is within the required base directory.
+     *
+     * @param requiredBase required base directory
+     * @param filePath file path to be tested
+     * @return the normalized file path is within the required base directory
```

Comment:
@return the normalized file path ***if*** within the required base directory

When I read it initially, my mind went to a boolean return.